src/test/java/spoon/reflect/visitor/DefaultJavaPrettyPrinterTest.java

```diff
@@ -150,4 +156,29 @@ void testPrintNewArrayWithImmediateAccess(String line) {
         CtType<?> first = model.getAllTypes().iterator().next();
         assertThat(first.toString(), containsString(line));
     }
+
+    @Test
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    void testPrintMethodReferenceTypeParameters() {
+        // contract: type parameters of method references are printed
+        Launcher launcher = new Launcher();
+        Factory factory = launcher.getFactory();
+        CtTypeReference<?> typeReference = factory.Type().createReference("Arrays");
+        CtExecutableReferenceExpression exeExpression = factory.createExecutableReferenceExpression();
+        exeExpression.setExecutable(factory.createExecutableReference());
+        exeExpression.getExecutable().setSimpleName("binarySearch");
+        exeExpression.setTarget(factory.createTypeAccess(typeReference));
+        // if no type parameters are given, no < and no > should be printed
+        assertThat(exeExpression.toString(), not(anyOf(containsString("<"), containsString(">"))));
+
+        exeExpression.getExecutable().addActualTypeArgument(factory.Type().integerType());
+        // with type parameters, the < and > should be printed
+        assertThat(exeExpression.toString(), allOf(containsString("<"), containsString(">")));
+        // the type parameter should appear
+        assertThat(exeExpression.toString(), containsString("Integer"));
+
+        exeExpression.getExecutable().addActualTypeArgument(factory.Type().integerType());
+        // more than one parameter type should be separated (with .* to allow imports)
+        assertThat(exeExpression.toString(), containsRegexMatch("<(.*)Integer, (.*)Integer>"));
```

Comment:
how about wrapping all of them in an [assertAll](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/Assertions.html#assertAll(org.junit.jupiter.api.function.Executable...))

Reply:
That would not work well with the current design of the test, as the steps between the asserts are important